Doc/library/os.rst

```diff
@@ -1195,6 +1238,47 @@ or `the MSDN <https://msdn.microsoft.com/en-us/library/z0kc8e3z.aspx>`_ on Windo
    .. versionadded:: 3.3
 
 
+.. function:: preadv(fd, buffers, offset, flags=0)
+
+   Combines the functionality of :func:`os.readv` and :func:`os.pread`. It
+   reads from a file descriptor *fd* into a number of mutable :term:`bytes-like
+   objects <bytes-like object>` *buffers*. As :func:`os.readv`, it will transfer
+   data into each buffer until it is full and then move on to the next buffer in
+   the sequence to hold the rest of the data. Its fourth argument, *offset*,
+   specifies the file offset at which the input operation is to be performed.
+   :func:`~os.preadv` return the total number of bytes read (which can be less than
+   the total capacity of all the objects).
+
+   The flags argument contains a bitwise OR of zero or more of the following
+   flags:
+
+       - RWF_HIPRI
+       - RWF_NOWAIT
+
+   :func:`~os.preadv` will call *preadv2* system call if available, which modifies
+   the behavior on a per-call basis based on the value of the *flags* argument. The
+   :c:func:`preadv2` function is required to pass flags.
+
+   *preadv2* and flags different than zero are available since Linux version
+   4.6 and newer.
+
+   Availability: Linux (version 2.6.30), FreeBSD 6.0 and newer,
+   OpenBSD (version 2.7 and newer).
+
+   .. versionadded:: 3.7
+
+
+.. data:: RWF_HIPRI
+          RWF_NOWAIT
```

Comment:
Hum, I suggest to have one ".. data" entry per constant and copy the documentation from the manual page. Extract of Fedora 27 manual pages:

```
       RWF_DSYNC (since Linux 4.7)
              Provide  a  per-write  equivalent  of  the O_DSYNC open(2) flag.
              This flag is meaningful only  for  pwritev2(),  and  its  effect
              applies only to the data range written by the system call.

       RWF_HIPRI (since Linux 4.6)
              High priority read/write.  Allows block-based filesystems to use
              polling of the device, which provides lower latency, but may use
              additional  resources.   (Currently, this feature is usable only
              on a file descriptor opened using the O_DIRECT flag.)

       RWF_SYNC (since Linux 4.7)
              Provide a per-write equivalent of the O_SYNC open(2) flag.  This
              flag  is  meaningful only for pwritev2(), and its effect applies
              only to the data range written by the system call.
```

It would become simpler to describe the availability.